lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -231,6 +241,98 @@ def __init__(self, *args):
 
 logger = logging.getLogger(__name__)
 
+RELATION_INTERFACE_NAME = "loki_push_api"
+
+
+class RelationDirection(Enum):
+    """Relations can be either provided or required by a charm."""
+
+    PROVIDED = 1
+    REQUIRED = 2
```

Comment:
This semantic already exists in `ops.charm.RelationRole`.
See alertmanager impl for reference
https://github.com/canonical/alertmanager-operator/blob/e7b4634b15b568fa96d2c07141d107ab801fce79/lib/charms/alertmanager_k8s/v0/alertmanager_dispatch.py#L81